docs/msbuild/customize-your-build.md

```diff
@@ -86,6 +98,51 @@ A summary of MSBuild's general approach is as follows:
 
 Or more simply: the first *Directory.Build.props* which doesn't import anything, is where MSBuild stops.
 
-## See Also  
- [MSBuild Concepts](../msbuild/msbuild-concepts.md)   
- [MSBuild Reference](../msbuild/msbuild-reference.md)   
+## MSBuildProjectExtensionsPath
+
+By default, `Microsoft.Common.props` imports `$(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.props` and `Microsoft.Common.targets` imports `$(MSBuildProjectExtensionsPath)$(MSBuildProjectFile).*.targets`. The default value of `MSBuildProjectExtensionsPath` is `$(BaseIntermediateOutputPath)`, `obj/`. This is the mechanism that NuGet uses to refer to build logic delivered with packages--at restore time, it creates `{project}.nuget.g.props` files that refer to the package contents.
+
+This extensibility mechanism can be disabled by setting the property `ImportProjectExtensionProps` to `false` before importing `Microsoft.Common.props`.
+
+## MSBuildExtensionsPath and MSBuildUserExtensionsPath
+
+> [!WARNING]
+> Using these extension mechanisms makes it harder to get repeatable builds across machines. Prefer configuration that can be checked into your source control system and shared among all developers of your codebase.
+
+By convention, many core build logic files import
+
+```
+$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\{TargetFileName}\ImportBefore\*.targets
+```
+
+before their contents, and
+
+```
+$(MSBuildExtensionsPath)\$(MSBuildToolsVersion)\{TargetFileName}\ImportAfter\*.targets
+```
+
+afterward. This allows installed SDKs to augment the build logic of common project types.
+
+The same directory structure is searched in `$(MSBuildUserExtensionsPath)`, which is the per-user folder `%LOCALAPPDATA%\Microsoft\MSBuild`. Files placed in that folder will be imported for all builds of the corresponding project type run under that user's credentials.
+
+## Customizing the solution build
+
+> [!IMPORTANT]
+> Customizing the solution build in this way applies only to command-line builds with `MSBuild.exe`. It **does not** apply to builds inside Visual Studio.
+
```

Comment:
Is this also true for directory props/targets files?

Reply:
It is _not_. VS builds import d.b.props/targets exactly the same way the command line does.

Reply:
cool!